pages/docs/tutorials/native/apple-framework.md

```diff
@@ -0,0 +1,284 @@
+
+Now we call the following commands to compile the code into Frameworks
+for macOS, iOS and iOS emulator respectively:
+```bash
+kotlinc lib.kt -produce dynamic -output macOS/Demo
+kotlinc lib.kt -produce framework -target ios_arm64 -output iOS/Demo
+kotlinc lib.kt -produce framework -target ios_x64 -output iOS_emu/Demo
+```
+
+The `kotlinc` generates three frameworks for us, named `Demo.framework` under 
+`macOS`, `iOS` and `iOS_emu` folders respectively.
+
+Let's see what is inside
+
+## Generated Framework Headers
+
+Each of created frameworks contains the header file in `<Framework>/Headers/Demo.h`.
+The file does not depend on the target platform (at least with Kotlin/Native v.0.8.1).
+It contains the definitions for our Kotlin code and few Kotlin-wide declarations. Let's
+see Kotlin runtime declarations first:
+
+<div class="sample" markdown="1" mode="obj-c" theme="idea" data-highlight-only auto-indent="false">
+
+```obj-c
+@interface KotlinBase : NSObject
+- (instancetype)init __attribute__((unavailable));
++ (instancetype)new __attribute__((unavailable));
++ (void)initialize __attribute__((objc_requires_super));
+@end;
+
+@interface KotlinBase (KotlinBaseCopying) <NSCopying>
+@end;
+
+__attribute__((objc_runtime_name("KotlinMutableSet")))
+@interface DemoMutableSet<ObjectType> : NSMutableSet<ObjectType>
+@end;
+
+__attribute__((objc_runtime_name("KotlinMutableDictionary")))
+@interface DemoMutableDictionary<KeyType, ObjectType> : NSMutableDictionary<KeyType, ObjectType>
+@end;
+
+@interface NSError (NSErrorKotlinException)
+@property (readonly) id _Nullable kotlinException;
+@end;
+```
+</div>
+
+Kotlin/Native classes have `KotlinBase` base class in Objective-C, the class extends
+the `NSObject` class there. We also have wrappers for collections and exceptions. 
+Most of collections are mapped to collections from the other side:
+
+|Kotlin|Swift|Objective-C|
+-------|-----|-----------|
+|List|Array|NSArray|
+|MutableList|NSMutableArray|NSMutableArray|
+|Set|Set|NSSet|
+|Map|Dictionary|NSDictionary|
+|MutableMap|NSMutableDictionary|NSMutableDictionary|
+{:.wide.zebra}
+
+The second part of the `<Framework>/Headers/Demo.h` file contains the exact definitions for our Kotlin declarations.
+The code is full of Objective-C attributes, which are intended to help
+using the Framework from both Objective-C and Swift languages. 
+
+<div class="sample" markdown="1" mode="obj-c" theme="idea" data-highlight-only="1" auto-indent="false">
+
+```obj-c
+@class DemoDemoObject, DemoDemoClazz;
+
+NS_ASSUME_NONNULL_BEGIN
+
+__attribute__((objc_subclassing_restricted))
+@interface DemoDemoObject : KotlinBase
+
++ (instancetype)alloc 
+      __attribute__((unavailable));
+      
++ (instancetype)allocWithZone:
+     (struct _NSZone *)zone 
+     __attribute__((unavailable));
+     
++ (instancetype)demoObject 
+     __attribute__((swift_name("init()")));
+     
+@property (readonly) NSString *field;
+@end;
+
+__attribute__((objc_subclassing_restricted))
```

Comment:
Exact generated bindings content may change without a warning, and we shalln't put too much emphasis on how what is inside, but on hw to use it.